### ISSUE TITLE
For loop is not working in the new release

### ISSUE THREAD

**Issue by ecations** [Scirra/Construct-3-bugs#4549]
## Problem description

For loop is not working in the new release. The loop is running but somehow the iterations are not adding up.

## Attach a .c3p

[loop bug further minimized r229.zip](https://github.com/Scirra/Construct-3-bugs/files/5949398/loop.bug.further.minimized.r229.zip)

## Steps to reproduce

1. Open file in r229 and play the game. Press "B". The game will build a row of bricks in the center.
2. Now open the file in r334 and do the same steps. The game fails at building a row of bricks (only two bricks appear).

## Observed result

The for loop behaves differently between versions.

## Expected result

A wall of bricks being built in the middle of the screen.

## More details

**Affected browsers/platforms:** 

Tested on Chrome in Windows 8.

**First affected release:** 

I know it works in r229 but not in r334.

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 88.0.4324.104
Browser engine: Chromium
Browser architecture: 64-bit
Context: browser
Operating system: Windows
Operating system version: 8.1
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 4
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.104 Safari/537.36
C3 release: r234.2 (stable)
Language setting: en-US

Local storage
Storage quota (approx): 134 gb
Storage usage (approx): 150 mb (0.1%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 970 Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
OVR_multiview2
Audio information
System sample rate: 44100 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
</details>

**Comment by jobel0092** [comment]
For Loop is working fine, the error is in the code inside the loop.

**Comment by ecations** [comment]
I did some more testing and can now confirm that this bug happens between r229 and r230. Something changes between those releases that breaks the function.

**Comment by jobel0092** [comment]
create an empty project a try to reproduce it with minimal objects and vars...

**Comment by ecations** [comment]
I did what you asked and created an empty file with just the relevant assets, variables and events.

**Comment by AshleyScirra** [comment]
In r230 we fixed the following bug:

> 'Wait' in function block did not handle local variables correctly

This refers to issue #4426 which we subsequently fixed. The problem was local variables inside function blocks not working correctly with the 'Wait' action. Your project combines all of these things. Presumably your events relied on the fact the bug was happening to work correctly; now we've fixed the bug, your events can no longer rely on the incorrect behavior, and so will need updating to work with the new correct behavior. I'm afraid there's nothing much else we can do about this: occasionally fixing bugs are not completely backwards compatible, and reintroducing the bug is not really a sensible thing to do, so you will have to update your project.
